torch/csrc/api/src/nn/init.cpp

```diff
@@ -36,6 +36,25 @@ struct Fan {
 };
 } // namespace
 
+TORCH_API double calculate_gain(Nonlinearity nonlinearity, double param) {
+  if (nonlinearity == Nonlinearity::linear ||
+     nonlinearity == Nonlinearity::conv1d ||
+     nonlinearity == Nonlinearity::conv2d ||
+     nonlinearity == Nonlinearity::conv3d ||
+     nonlinearity == Nonlinearity::conv_transpose1d ||
+     nonlinearity == Nonlinearity::conv_transpose2d ||
+     nonlinearity == Nonlinearity::conv_transpose3d ||
+     nonlinearity == Nonlinearity::sigmoid) {
+       return 1.0;
+  } else if (nonlinearity == Nonlinearity::tanh) {
+    return 5.0 / 3.0;
+  } else if (nonlinearity == Nonlinearity::relu) {
+    return sqrt(2.0);
+  } else if (nonlinearity == Nonlinearity::leaky_relu) {
+    return sqrt(2.0 / (1 + pow(param, 2)));
+  }
```

Comment:
Should I add an `else` here? In the future if we add a new `Nonlinearity` but don't update this code we wouldn't be returning anything from this method. What's the proper pattern here? Throw an error?

Reply:
I would say remove the huge if block above and return `1.0` at the bottom

Reply:
Good idea, I've made this change.